### ISSUE TITLE
Migrate to Single package.json structure

### ISSUE THREAD

**Issue by deiucanta** [SimulatedGREG/electron-vue#112]
There are two `package.json`'s right now and I read the docs on why it is like that. But I just noticed that the root one uses only dev dependencies and the other one uses "normal" dependencies. If you were to merge them into a single one there would be an easy way to tell which dependency is for what.

Am I missing something?

**Comment by SimulatedGREG** [comment]
@deiucanta 

Yes you could separate dependencies using `devDependencies` but then `electron-packager` wouldn't know the different and also include all those `devDependencies` into your production builds where they will never be used. In terms of `electron-builder`, this is actually possible.

**Comment by deiucanta** [comment]
@SimulatedGREG thanks

Are you aware if `electron-packager` is planning to do something about it? The overhead is not that big when working with two `package.json` files but if we can improve the developer experience... why not?

**Comment by citycide** [comment]
@deiucanta you'd want to check their issues / projects for that. Like @SimulatedGREG said though, `electron-builder` ( which this boilerplate provides ) supports the single-manifest setup you're asking for.

**Comment by SimulatedGREG** [comment]
Well it turns out `electron-packager` did release make an update to `--prune` `devDependencies`. This is something that was not available at the initial creation of this boilerplate. Its worth considering, but IMO I feel a separate `app` folder is a much easier structure to use. Typically when you put all source files and dev configurations in one space its gets cluttered quickly.

**Comment by develar** [comment]
electron-builder doesn't require you to use separate directories. Only production deps are packed and only production native deps are rebuild. And [files](https://github.com/electron-userland/electron-builder/wiki/Options#Config-files)/extraFiles/extraResources allows you to easily configure files to pack.

**Comment by SimulatedGREG** [comment]
I'm fully aware now that both packagers handle this properly, but I'm still not convinced putting everything in one document space really keeps things organized. This goes probably into a bigger discussion about whether or not its worth putting **all** dev configuration files into its own folder (maybe `.electron-vue`?).

**Comment by citycide** [comment]
Here's the other side of the coin though - in this ecosystem there's already a lot of muscle memory involved in installing dependencies. In a node app the root project folder is the only place we install things. We've already got a clear separation within `package.json` between `dependencies` and `devDependencies`.

Separating them further into two files kind of flies in the face of the ecosystem, and I think the electron packaging tools acknowledged this to some extent by supporting the single file structure.

TL:DR; there's a reason both major packaging tools now support single file structure and a reason there's already been issues raised because people installed a dependency in the wrong folder.

**Comment by SimulatedGREG** [comment]
Just as a quick experiment, putting all configurations in a folder `.electron-vue` does clean up nicely. What do you guys think?

![image](https://cloud.githubusercontent.com/assets/8327811/22855958/0238d636-f04e-11e6-9e12-de910593e9fe.png)

## Expanded
![image](https://cloud.githubusercontent.com/assets/8327811/22855962/152d6856-f04e-11e6-9139-fb25ce76d42b.png)

**Comment by SimulatedGREG** [comment]
Going to spend some time today to implement this. I'll have an update when you guys can give it a try on another branch.

**Comment by deiucanta** [comment]
looks good :) let me know when you have something up

**Comment by SimulatedGREG** [comment]
Currently on vacation, but got some free time. You can now try the first draft of this new structure with...

```bash
vue init simulatedgreg/electron-vue#feat/simpler-structure my-project
```

Currently have not brought support for all `vue-cli` configurations expect for the defaults. Most other configurations should work fine, but the major one that still needs work is `electron-builder`.

**Comment by rafaelpimpa** [comment]
I'm using the ``simpler-structure`` on a new project and It's actually great, didn't have any problems until now.

**Comment by Derek-X-Wang** [comment]
@SimulatedGREG 
I created a new project by
`vue init simulatedgreg/electron-vue#feat/simpler-structure my-project`

However, npm run lint:fix triggers this error:

<img width="1091" alt="screen shot 2017-03-04 at 3 04 18 am" src="https://cloud.githubusercontent.com/assets/6364170/23578589/74655cf6-008f-11e7-8faf-4fe39f3e1334.png">

And my editor(Atom)'s eslint says “Parsing error: The keyword ‘import’ is reserved.” 

It seems that .eslintrc.js cannot be found(by Atom?) within .electron-vue.

In my case, moving .eslintrc.js to the root directory solves eslint problem.
package.json
line 11: `"lint": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter --ignore-path .electron-vue/.eslintignore src test",`
line 12: `"lint:fix": "eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter --fix --ignore-path .electron-vue/.eslintignore src test",`

.electron-vue/webpack.renderer.config.js
line 128: `configFile: path.resolve(__dirname, '../.eslintrc.js'),`

Is this a bug or I miss some configuration?

**Comment by SimulatedGREG** [comment]
I was hoping to move .config files into the `.electron-vue` folder but that doesn't seems like it will be feasible for users that have text editors expecting them to be in the root directory.

**Comment by Derek-X-Wang** [comment]
Well, it seems only .eslintrc has to be in the root directory. I leave all other .config files in .electron-vue and it works fine.
Overall, I still think it's a great structure. Like it a lot :smiley:

**Comment by alexey2baranov** [comment]
I like new structure very much too. Waiting stable release

**Comment by SimulatedGREG** [comment]
#171

**Comment by SimulatedGREG** [comment]
Possible, but I would like to keep everything in that folder if possible. What benefit would there be for that?

**Comment by re-fort** [comment]
I just felt it is unnecessary to separate it from the directory where `.babelrc` and `.eslintrc.js` are placed, but there is no further reason.

**Comment by SimulatedGREG** [comment]
Alright, I believe everything has been migrated over to a new single `package.json` setup and merged to the `dev` branch. There may still be a few issues here and there, so please do provide any feedback if you run into anything.
